dashboard/15-final/app/api/auth/[...nextauth]/route.ts

```diff
@@ -0,0 +1,53 @@
+import NextAuth, { NextAuthOptions } from 'next-auth';
+import CredentialsProvider from 'next-auth/providers/credentials';
+import bcrypt from 'bcrypt';
+import { fetchUser } from '../../../lib/data';
+import { User } from '@/app/lib/definitions';
+
+const authOptions: NextAuthOptions = {
+  providers: [
+    CredentialsProvider({
+      name: 'credentials',
+      credentials: {},
```

Comment:
```suggestion
      credentials: {
        password: { label: "Password", type: "password" },
        email: { label: "Email", type: "email" },
      },
```

This will type `authorize` correctly.